unison-cli/src/Unison/Cli/MonadUtils.hs

```diff
@@ -144,18 +152,34 @@ resolveShortBranchHash hash = do
     pure (fromMaybe Branch.empty branch)
 
 ------------------------------------------------------------------------------------------------------------------------
--- Getting branches
+-- Getting/Setting branches
 
 -- | Get the root branch.
 getRootBranch :: Cli r (Branch IO)
 getRootBranch = do
-  use #root
+  use #root >>= atomically . readTMVar
 
 -- | Get the root branch0.
 getRootBranch0 :: Cli r (Branch0 IO)
 getRootBranch0 =
   Branch.head <$> getRootBranch
 
+-- | Set a new root branch.
+-- Note: This does _not_ update the codebase, the caller is responsible for that.
+setRootBranch :: Branch IO -> Cli r ()
+setRootBranch b = do
+  void $ modifyRootBranch (const b)
+
+-- | Get the root branch.
+modifyRootBranch :: (Branch IO -> Branch IO) -> Cli r (Branch IO)
+modifyRootBranch f = do
+  rootVar <- use #root
+  atomically do
+    root <- takeTMVar rootVar
+    let newRoot = f root
+    putTMVar rootVar newRoot
```

Comment:
I think we probably want a strict 'put' here

Reply:
There doesn't appear to be a strict `putTMVar`; would you like me to force with `$!`?